lib/data_hygiene/document_status_checker.rb

```diff
@@ -0,0 +1,40 @@
+class DataHygiene::DocumentStatusChecker
+  def initialize(document)
+    @document = document
+  end
+
+  def content_store?
+    routes.each do |route|
+      content_item = GdsApi.content_store.content_item(route[:path])
+      updated_at = DateTime.parse(content_item["updated_at"])
+      return false unless updated_at > edition.published_at # content-store must be later due to latency
+    rescue GdsApi::ContentStore::ItemNotFound
+      return false
+    end
+
+    true
+  end
+
+  def router?
+    routes.each do |route|
+      route = GdsApi.router.get_route(base_path)
+      return false unless route["backend_id"] == edition.rendering_app
+    rescue GdsApi::ContentStore::ItemNotFound
```

Comment:
I'm not sure if it makes any real difference but because you're `return`ing out of this loop you could pull the exception handling up to the level of the `router?` method rather than having it defined within the loop.